[Docs] Enhance AG Grid documentation with column sizing options (#1306)
mckinsey/vizro: vizro-core/docs/pages/user-guides/table.md

No specific formatting is available for custom objects and strings, however you can make use of [Value Formatters](https://dash.plotly.com/dash-ag-grid/value-formatters) to format displayed strings automatically.

### Column Size

> The AG Grid component provides automatic column sizing options through the `columnSize` property. This feature allows you to control how columns are sized within the grid to optimize the display of your data.

Reviewer: ```suggestion
The AgGrid provides automatic column sizing options through the `columnSize` property. This feature allows you to control how columns are sized within the grid to optimize the display of your data.
```

> You can configure column sizing by setting the `columnSize` parameter in your `dash_ag_grid` function call. The available options are:

Reviewer: ```suggestion
You can configure column sizing by setting the `columnSize` parameter in your `dash_ag_grid` function call. By default, the AgGrid is set to `sizeToFit` within vizro. The available options are:
```

- **`"autoSize"`**: Automatically adjusts column widths to fit their content. This is particularly useful when you have varying content lengths and want each column to be sized appropriately for readability.

- **`"sizeToFit"`**: Resizes all columns proportionally to fill the entire width of the grid container. This ensures no horizontal scrolling is needed and the table utilizes the full available space.

> - **`"responsiveSizeToFit"`**: Similar to `sizeToFit`, but also automatically readjusts column sizes when the grid container or columns change (such as when the browser window is resized or when filters are applied).
> 

Reviewer: ```suggestion
- **`"responsiveSizeToFit"`**: Combines `sizeToFit` with automatic readjustment of the columns' widths when the grid container or columns change (such as when the browser window is resized or when filters are applied).
```

> - **`None`** (default): Maintains the default column widths without automatic resizing.

Reviewer: ```suggestion
- **`None`**: Maintains the default column widths without automatic resizing.
```

Author: Yea, forgot to remove this one.  Thanks for the extra pair of 👀

For more advanced column sizing configurations, you can use the `columnSizeOptions` parameter in combination with `columnSize`.

!!! example "AG Grid with column sizing"

    === "app.py"

        ```
        import vizro.models as vm
        import vizro.plotly.express as px
        from vizro import Vizro
        from vizro.tables import dash_ag_grid

        df = px.data.gapminder()

        page = vm.Page(
            title="AG Grid with Column Sizing",
            components=[
                vm.AgGrid(
                    title="AG Grid with auto-sized columns",
                    figure=dash_ag_grid(data_frame=df, columnSize="autoSize")
                )
            ]
        )

        dashboard = vm.Dashboard(pages=[page])
        Vizro().build(dashboard).run()
        ```

    ```

>     ```

Reviewer: This is one of the annoying parts, but could you add a yaml example and a screenshot here? Take a look at other docs examples :)
